nightly_tests/nightly_tests_ondemand/conftest.py

```diff
@@ -1,31 +1,44 @@
 import pytest
 
+# Fixture for initializing an empty list to store test results. Scope is set to 'session' to show across all tests in a session.
 @pytest.fixture(scope="session")
 def test_results(request):
     request.session.results = []
     yield request.session.results
 
+# A hook implementation to intercept test reports. This function records the name and status of each test when it's called.
 @pytest.hookimpl(tryfirst=True, hookwrapper=True)
 def pytest_runtest_makereport(item, call):
     outcome = yield
     result = outcome.get_result()
     
+    # Record the test name and its pass/fail status after each test is executed.
     if result.when == 'call':
         test_name = item.name
         status = 'PASSED' if result.passed else 'FAILED'
         item.session.results.append({'name': test_name, 'status': status})
 
+# Function executed at the end of the test session. It writes the test results to a file.
 def pytest_sessionfinish(session, exitstatus):
     with open('makereport_output.txt', 'w') as f:
         print_table(session.results, file=f)
 
+# Function to print the test results in a table format. The output can be directed to a file.
 def print_table(results, file=None):
+    # ASCII symbols for pass and fail
+    pass_symbol = '✔'
```

Comment:
Minor comment.  Usually I'm all for variable use, if it's used in multiple places, and can optimize runtime/memory.  However, as far as I see, these are only used on line 42 below.  I would rather just see these ascii symbols inline on line 42 for readability, and just get rid of lines 29 and 30.

Reply:
Agreed, makes sense. Incorporating it directing in its specific context will improve readability.